### PR TITLE
obs-studio: 26.0.2 -> 26.1.2

### PR DESCRIPTION
**Pull request by lambdadog**
###### Motivation for this change

Bump version. Allows using the new Twitch "VOD Audio Channel".

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111029` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>obs-ndi: log was empty</li>
  </ul>
</details>
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>obs-gstreamer</li>
<li>obs-move-transition</li>
<li>obs-studio</li>
<li>obs-v4l2sink</li>
<li>obs-wlrobs</li>
</ul>
</details>

File: pkgs/applications/video/obs-studio/default.nix [hunk 1]
```diff
@@ -40,13 +40,13 @@ let
 
 in mkDerivation rec {
   pname = "obs-studio";
-  version = "26.0.2";
+  version = "26.1.2";
 
   src = fetchFromGitHub {
     owner = "obsproject";
     repo = "obs-studio";
     rev = "refs/tags/${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/video/obs-studio/default.nix:48]:
```suggestion
    rev = version;
```

**Comment by lambdadog** [comment]
`obs-ndi` builds after my changes in #111048, although I'm uncertain whether an automated `nixpkgs-review` will be able to confirm that seeing as the NDI SDK download is a manual process.

**Comment by Munksgaard** [comment]
@lambdadog Did you check that the browser plugin still works? I see that you removed the use of `refs/tags/${version}` as the rev, but I think that has the unfortunate effect of not pulling submodules, which in turn means that the browser plugin does not work.

**Comment by SuperSandro2000** [comment]
According to the log it build and linked the browser plugin.

**Comment by Munksgaard** [comment]
Okay, thank you for checking. I'm having (unrelated) problems with my local obs install, so I couldn't verify it myself.

**Pull request merged**